hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheOperation.java

```diff
@@ -73,6 +77,11 @@ public final void beforeRun() throws Exception {
         cacheService = getService();
         try {
             recordStore = getOrCreateStoreIfAllowed();
+            // establish tenant application's thread-local context for this cache operation
+            CacheConfig<?, ?> cacheConfig = cacheService.getCacheConfig(name);
```

Comment:
can't we get this config from recordstore? If we can, we eliminate concurrent hashmap lookup cost.

Reply:
good catch!